src/Service/EnrichOrganizationService.php

```diff
@@ -429,6 +361,104 @@ public function enrichGithubOrganization(ObjectEntity $organization): ObjectEnti
     }//end enrichGithubOrganization()
 
 
+    /**
+     * This function gets all the repositories from the given organization and sets it to the owns of the organization.
+     *
+     * @param ObjectEntity $organization Catalogi organization https://opencatalogi.nl/oc.organisation.schema.json
+     *
+     * @throws GuzzleException|Exception
+     *
+     * @return ObjectEntity
+     */
+    public function enrichGitlabOrganization(ObjectEntity $organization): ObjectEntity
+    {
+        // TODO: Get the gitlab reference from the configuration array.
+        // Get the gitlab api source.
+        $source = $this->resourceService->getSource('https://opencatalogi.nl/source/oc.GitlabAPI.source.json', 'open-catalogi/open-catalogi-bundle');
+        if ($source === null
+            || $this->gitlabApiService->checkGitlabAuth($source) === false
+        ) {
+            return $organization;
+        }//end if
+
+        // Get the path of the gitlab url.
+        $gitlabPath = \Safe\parse_url($organization->getValue('gitlab'))['path'];
+
+        if ($organization->getValue('type') === 'Organization') {
+            $gitlabPath = explode('/groups/', $gitlabPath)[1];
+            $gitlabPath = urlencode($gitlabPath);
+
+            // Get the group from the gitlab api.
+            $organizationArray = $this->gitlabApiService->getOrganization($gitlabPath, $source);
+        }
+
+        // TODO: Test this case.
+        if ($organization->getValue('type') === 'User') {
+            // Get the user from the github api.
+            $organizationArray = $this->gitlabApiService->getUser($gitlabPath, $source);
+        }
+
+        // Update the organization with the opencatalogi file.
+        $opencatalogiRepo = $organization->getValue('opencatalogiRepo');
+        $opencatalogiRepo = 'https://gitlab.com/api/v4/projects/33855802/repository/files/publiccode.yml?ref=master';
+
+        if ($opencatalogiRepo !== null) {
+            $parsedUrl = \Safe\parse_url($opencatalogiRepo);
+
+            // Get the default_branch from the parsed url query.
+            $repositoryArray['default_branch'] = explode('ref=', $parsedUrl['query'])[1];
+
+            $explodedPath = explode('/api/v4/projects/', $parsedUrl['path'])[1];
+            $explodedPath = explode('/repository/files/', $explodedPath);
+
+            // Get the id and path from the parsed url path.
+            $repositoryArray['id'] = $explodedPath[0];
+            $directory['path']     = $explodedPath[1];
```

Comment:
Looks like this should work fine with your hardcoded url.
Something for in the future to look into: 
https://www.php.net/manual/en/function.preg-match.php 

https://regex101.com/ 
With regex you can create 1 regex and find all properties from your URL in a few lines of code. (regex should be faster as well)